RecoVertex/PrimaryVertexProducer/plugins/PrimaryVertexProducer.cc

```diff
@@ -160,6 +160,37 @@ void PrimaryVertexProducer::produce(edm::Event& iEvent, const edm::EventSetup& i
   edm::Handle<reco::TrackCollection> tks;
   iEvent.getByToken(trkToken, tks);
 
+  // mechanism to put the beamspot if the track collection is empty
+  if (!tks.isValid()) {
+    for (std::vector<algo>::const_iterator algorithm = algorithms.begin(); algorithm != algorithms.end(); algorithm++) {
+      auto result = std::make_unique<reco::VertexCollection>();
+      reco::VertexCollection& vColl = (*result);
+
+      GlobalError bse(beamSpot.rotatedCovariance3D());
+      if ((bse.cxx() <= 0.) || (bse.cyy() <= 0.) || (bse.czz() <= 0.)) {
+        AlgebraicSymMatrix33 we;
+        we(0, 0) = 10000;
+        we(1, 1) = 10000;
+        we(2, 2) = 10000;
+        vColl.push_back(reco::Vertex(beamSpot.position(), we, 0., 0., 0));
+        if (fVerbose) {
+          std::cout << "RecoVertex/PrimaryVertexProducer: "
```

Comment:
use `std::cout` here, not Log?

Reply:
@srimanob `std::cout` is used throughout this class when the `fVerbose` flag is active (false by default). I agree it should be moved to log messaging, but then it should be done for the whole class. I can take care of that once this PR and the other PR affecting this class (https://github.com/cms-sw/cmssw/pull/43592) get merged (IF the maintainers agree).

Reply:
This is kind of the same with previous review of ECAL code (collect part of code cleaing https://github.com/cms-sw/cmssw/issues/43515#issuecomment-1852701072). I am fine that this will be clean up later.